frontend/apps/reader/modules/readerbookmark.lua

```diff
@@ -292,56 +366,262 @@ function ReaderBookmark:onShowBookmark()
         bm_menu,
     }
 
-    -- buid up menu widget method as closure
+    -- build up menu widget method as closure
     local bookmark = self
     function bm_menu:onMenuChoice(item)
         bookmark.ui.link:addCurrentLocationToStack()
         bookmark:gotoBookmark(item.page)
     end
 
     function bm_menu:onMenuHold(item)
+        -- reverse the bookmarks (because KOReader bookmarks are sorted in reverse of natural page order):
+        -- not sure whether this is always the case?
+        local bookmarks = {}
+        for i = #bookmark.bookmarks, 1, -1 do
+            local temp = bookmark.bookmarks[i]
+            -- we need this reference to the unreversed, original bookmarks for deleting or renaming them:
+            temp.index = i
+            table.insert(bookmarks, temp)
+        end
+        local current_bookmark = 1
+        local page = ""
+        for nr, bmk in ipairs(bookmarks) do
+            if item.index == bmk.index then
+                current_bookmark = nr
+                break
+            end
+        end
+        local title = T(_("Bookmark details  -  %1/%2"), tostring(current_bookmark), tostring(#bookmarks))
+        -- show page of bookmark in title:
+        local ipage = item.page
+        if not bookmark.ui.document.info.has_pages then
+            page = bookmark.ui.document:getPageFromXPointer(ipage)
+        end
+        if page ~= "" then
+            title = title .. "  -  " .. _("page") .. " " .. page
+        end
+        -- show search term in title:
+        if bookmark.search_value and bookmark.search_value ~= "" then
+            title = title .. '  -  "' .. bookmark.search_value .. '"'
```

Comment:
These 2 title = might be more readable if using `T('%1 - "%2"')`. Dunno if these constructs need to be translatable.